delira/models/backends/tf_eager/data_parallel.py

```diff
@@ -0,0 +1,48 @@
+import tensorflow as tf
+from delira.models.backends.tf_eager.abstract_network import \
+    AbstractTfEagerNetwork
+
+
+class DataParallelTfEagerNetwork(AbstractTfEagerNetwork):
+    """
+    DataParallel Module for the TF eager execution backend
+    """
+
+    def __init__(self, module, devices):
+        """
+
+        Parameters
+        ----------
+        module : :class:`AbstractTfEagerNetwork`
+            the module to scatter across different devices
+        devices : list
+            list of ints specifying the GPU indices
+        """
+        super().__init__()
+
+        self._closure = module.closure
+        self._prepare_batch = module.pepare_batch
+
+        self.module = tf.keras.utils.multi_gpu_model(module, devices, True)
```

Comment:
This is most likely not going to work (it only works when using high-level Keras APIs, e.g. self.module.fit(), see [here](https://github.com/tensorflow/tensorflow/issues/21788) ).

I suggest open a [FR/look at where the tf.distribute support goes for custom training loops](https://www.tensorflow.org/beta/guide/distribute_strategy#using_tfdistributestrategy_with_custom_training_loops)  and enforcing single-GPU mode in between

Reply:
I disabled this for now and will track this in #177